docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+    A2 --> A3
+
+    subgraph getAgent ["getAgent()"]
+      A3["Check for DesktopAgentDetails in SessionStorage"] --> P1{"Does window.fdc3 exist?"}
+      P1 -->|yes|P2["stop timeout"]
+      P2 --> P21["Save DesktopAgentDetails to SessionStorage"]
+      P1 -->|No|P3["Listen for fdc3Ready"]
+      P3 --> P31["fdc3Ready event fires"]
+      P31 --> P32["stop timeout"]
+      P32 --> P33["Save DesktopAgentDetails to SessionStorage"]
+      
+      A3 --> B1{"Do parent refs exist?"}
+      B1 -->|yes|B11["Send WCP1Hello to all candidates"]
+      B11 --> B2["Receive WCP2LoadUrl"]
+      B2 --> B21["stop timeout"]
+      B21 --> B22["Create hidden iframe with URL"]
+      B22 --> B23["await iframe's load event"]
+      B23 --> B24["Send WCP1Hello to iframe"]
+      B24 --> B3["Receive WCP3Handshake with MessagePort"]
+      B3 --> B31["stop timeout"]
+      B11 --> B3
+      B31 --> B32["Send WCP4ValidateIdentity on MessagePort"]
+      B32 --> B321["Receive WCP5ValidateIdentityResponse"]
+      B321 --> B3211["Create DesktopAgentProxy to process DACP messages"]
+      B3211 --> B3212["Save DesktopAgentDetails to SessionStorage"]
+      B32 --> B322["Receive WCP5ValidateIdentityFailedResponse"]
+      
+      A3 --> T1["Set timeout"]
+      T1 --> T2["timeout expires"]
+      T2 --> T3{"Was a failover fn provided"}
+      T3 -->|yes|T31["Run failover"]
+      T31 --> T311{"Check failover return type"}
+      T311 -->|WindowProxy|T3111["Send WCP1Hello via WindowProxy"]
+      T311 -->|DesktopAgent|T3112["Save DesktopAgentDetails to SessionStorage"]
+      T3111 --> B3
+    end
+    P21 -->P22(["resolve with window.fdc3"])
+    P33 -->P34(["resolve with window.fdc3"])
+    B3212 --> B3213(["resolve with DesktopAgentProxy"])
+    B322 --> B3221(["reject with AgentError.AccessDenied"])
+    T3112 --> T31121(["resolve with DesktopAgent"])
+    T3 -->|no|T32(["reject with AgentError.AgentNotFound"])
+    T311 -->|other|T3113["reject with AgentError.InvalidFailover"]
+```
+
+## Providing Channel Selector and Intent Resolver UIs
+
+Users of FDC3 Desktop Agents often need access to UI controls that allow them to select user channels or to resolve intents that have multiple resolution options. Whilst apps can implement these UIs on their own via data and API calls provided by the `DesktopAgent` API, Desktop Agents typically provide these interfaces themselves.
+
+However, Browser Resident Desktop Agents may have difficulty displaying user interfaces over applications for a variety of reasons (inability to inject code, lack of permissions to display popups etc.), or may not (e.g. because they render applications in iframes within windows they control and can therefore display content over the iframe). The Web Connection Protocol and the `getAgent()` implementation based on it and incorporated into apps via the [`@finos/fdc3` npm module](https://www.npmjs.com/package/@finos/fdc3), is intended to help Desktop Agents deliver these UIs where necessary.
```

Comment:
- Missing comma before "etc." - "lack of permissions to display popups etc." should be "lack of permissions to display popups, etc."
- Could also consider adding a third reason before the comma - "user gestures not following cross-origin comms")

Reply:
fixed in 976af80d5ba8d831e202aabcd37d1a7886e4c930